instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,311 @@
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForEmptyFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testOneBillingSessionSentForPausedAndResumedFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionsForMixedFreeDriveAndActiveGuidanceTrip() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(3, billingRequests.size)
+        assertTrue(billingRequests[0].isFreeDriveSession)
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests[0].duration)
+        assertTrue(billingRequests[1].isActiveGuidanceSession)
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests[1].duration)
+        assertTrue(billingRequests[2].isFreeDriveSession)
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests[2].duration)
+    }
```

Comment:
Which option do you find easier to read and understand? 
```suggestion
    fun testBillingSessionsForMixedFreeDriveAndActiveGuidanceTrip() = sdkTest {
        val observer = HttpServiceEventsObserver.register()

        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
        delay(1000)
        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
        delay(1000)
        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
        delay(1000)
        mapboxNavigation.onDestroy()

        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)

        val billingRequests = observer.billingRequests()
        assertEquals(3, billingRequests.size)
        assertTrue(billingRequests[0].isFreeDriveSession)
        assertSessionDuration(1, billingRequests[0].duration)
        assertTrue(billingRequests[1].isActiveGuidanceSession)
        assertSessionDuration(1, billingRequests[1].duration)
        assertTrue(billingRequests[2].isFreeDriveSession)
        assertSessionDuration(1, billingRequests[2].duration)
    }
```
PS. I'm just sharing my opinion. I'm totally fine if you leave option you like the most 🙂

Reply:
Personally, I like constants more, just because of my habit of extracting constants :) As I see, we have different approaches in our tests. But I'm fine with any option, we can agree on a common approach and code style for tests. @VysotskiVadim @dzinad what do you think?